Assets/DemoScript.cs

```diff
@@ -0,0 +1,276 @@
+using System.Numerics;
+using Cysharp.Threading.Tasks;
+using MirageSDK.Core.Implementation;
+using MirageSDK.Core.Infrastructure;
+using MirageSDK.Core.Utils;
+using MirageSDK.Examples.ContractMessages.ERC1155;
+using MirageSDK.Examples.ContractMessages.GameCharacterContract;
+using MirageSDK.Examples.WearableNFTExample;
+using MirageSDK.WalletConnectSharp.Unity;
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class DemoScript : MonoBehaviour
+{
+	private const string TransactionGasLimit = "1000000";
+	private const string BlueHatAddress = "0x00010000000000000000000000000000000000000000000000000000000001";
+	private const string RedHatAddress = "0x00010000000000000000000000000000000000000000000000000000000002";
+
+	[SerializeField]
+	private GameObject _character;
+
+	[SerializeField]
+	private TMP_Text _characterID;
+
+	[SerializeField]
+	private GameObject _blueHat;
+
+	[SerializeField]
+	private Button _blueHatButton;
+
+	[SerializeField]
+	private GameObject _redHat;
+
+	[SerializeField]
+	private Button _redHatButton;
```

Comment:
you have to look into more generic solutions, because right now if you will have to add another type of item you will have to duplicate a LOT of code, instead of just adding the specifications